pkg/auth/rhsso_authz_handler_test.go

```diff
@@ -290,6 +291,230 @@ var _ = Describe("OwnedBy", func() {
+			payload := &ocm.AuthPayload{Username: "user2", Organization: "org1"}
+			ctx = context.WithValue(ctx, restapi.AuthKey, payload)
+
+			mockOcmAuthorization.EXPECT().AccessReview(
+				gomock.Any(), "user2", gomock.Any(), subscription.String(), gomock.Any()).
+				Return(true, nil).Times(2)
+			infraEnv, _ := common.GetInfraEnvFromDB(db, id1)
+			Expect(authzHandler.HasAccessTo(ctx, infraEnv, ReadAction)).To(BeTrue())
+			Expect(authzHandler.HasAccessTo(ctx, infraEnv, UpdateAction)).To(BeTrue())
+			Expect(authzHandler.HasAccessTo(ctx, infraEnv, DeleteAction)).To(BeTrue())
+			Expect(authzHandler.client.Cache.ItemCount()).To(Equal(2))
+
+			By("unbound infra-env rules according to infra-env")
+			infraEnv, _ = common.GetInfraEnvFromDB(db, id2)
+			Expect(authzHandler.HasAccessTo(ctx, infraEnv, ReadAction)).To(BeTrue()) //user2 has access rights because of the org
+			Expect(authzHandler.HasAccessTo(ctx, infraEnv, UpdateAction)).To(BeFalse())
+			Expect(authzHandler.HasAccessTo(ctx, infraEnv, DeleteAction)).To(BeFalse())
+			Expect(authzHandler.client.Cache.ItemCount()).To(Equal(2)) //nothing was added to the cache
+		})
+
+		It("others can read but not write", func() {
+			payload := &ocm.AuthPayload{Username: "user2", Organization: "org1"}
+			ctx = context.WithValue(ctx, restapi.AuthKey, payload)
+
+			mockOcmAuthorization.EXPECT().AccessReview(
+				gomock.Any(), "user2", gomock.Any(), subscription.String(), gomock.Any()).
+				Return(false, nil).Times(2)
+			infraEnv, _ := common.GetInfraEnvFromDB(db, id1)
+			Expect(authzHandler.HasAccessTo(ctx, infraEnv, ReadAction)).To(BeTrue()) //org based access
+			Expect(authzHandler.HasAccessTo(ctx, infraEnv, UpdateAction)).To(BeFalse())
+			Expect(authzHandler.HasAccessTo(ctx, infraEnv, DeleteAction)).To(BeFalse())
+			Expect(authzHandler.client.Cache.ItemCount()).To(Equal(2))
+		})
+	})
+
+	Context("host", func() {
+		It("owner can read and write according to infra-env ownership", func() {
+			payload := &ocm.AuthPayload{}
+			payload.Username = "user1"
+			payload.Organization = "org1"
+			ctx = context.WithValue(ctx, restapi.AuthKey, payload)
+
+			host, _ := common.GetHostFromDBbyHostId(db, id1)
+			Expect(authzHandler.HasAccessTo(ctx, host, ReadAction)).To(BeTrue())
+			Expect(authzHandler.HasAccessTo(ctx, host, UpdateAction)).To(BeTrue())
+			Expect(authzHandler.HasAccessTo(ctx, host, DeleteAction)).To(BeTrue())
+		})
+
+		It("ClusterEditorRole on cluster-bound host can read and write according to cluster rules", func() {
+			payload := &ocm.AuthPayload{Username: "user2", Organization: "org1"}
+			ctx = context.WithValue(ctx, restapi.AuthKey, payload)
+
+			//host[id2] bound to cluster[id1]
+			host, _ := common.GetHostFromDBbyHostId(db, id2)
+			mockOcmAuthorization.EXPECT().AccessReview(
+				gomock.Any(), "user2", gomock.Any(), "other", gomock.Any()).
+				Return(true, nil).Times(2)
+			Expect(authzHandler.HasAccessTo(ctx, host, ReadAction)).To(BeFalse())
```

Comment:
If update/delete action are allowed, shouldn't read action also be allowed here?

Reply:
It is actually a contrived example where the user is not the object owner but the access review allow them a write access. It will not actually happen in real life but it covers a code branch